### PR TITLE
Feature_fingerprint

### PR DESCRIPTION
**Pull request by BaojunCZ**
App will not ask user to authenticate fingerprint when user leave app within 20s

### CODE REVIEW

File: app/src/main/java/org/nervos/neuron/activity/BaseActivity.java [hunk 1]
```diff
@@ -62,9 +64,13 @@ private boolean isShouldTimeOut() {
     @Override
     protected void onResume() {
         super.onResume();
-        if (isShouldTimeOut() && needLogin) {
-            gotoLogin();
-            needLogin = false;
+        if (isShouldTimeOut()) {
```
Comment (ashchan) [inline app/src/main/java/org/nervos/neuron/activity/BaseActivity.java]: We could use `shouldTimeOut` or `isTimeOut`.

File: app/src/main/java/org/nervos/neuron/activity/BaseActivity.java [hunk 2]
```diff
@@ -98,21 +104,18 @@ private void gotoLogin() {
...
-        if (dialog == null)
-            dialog = new ProgressingDialog(this);
+        if (dialog == null) dialog = new ProgressingDialog(this);
         dialog.show();
     }
 
     protected void showProgressBar(@StringRes int message) {
-        if (dialog == null)
-            dialog = new ProgressingDialog(this);
+        if (dialog == null) dialog = new ProgressingDialog(this);
```
Comment (ashchan) [inline app/src/main/java/org/nervos/neuron/activity/BaseActivity.java]: Should always have `{}` even if there's only one statement in the block.

Comment (duanyytop) [reply app/src/main/java/org/nervos/neuron/activity/BaseActivity.java]: image

File: app/src/main/java/org/nervos/neuron/activity/BaseActivity.java [hunk 3]
```diff
@@ -23,24 +24,25 @@
 
 public class BaseActivity extends AppCompatActivity {
 
+    private static final int timeInterval = 20000;
```
Comment (duanyytop) [inline app/src/main/java/org/nervos/neuron/activity/BaseActivity.java]: 常量一般都是全大写的

**Pull request merged**